types/mongoose/index.d.ts

```diff
@@ -2550,7 +2550,7 @@ declare module "mongoose" {
      * @param options.validateBeforeSave set to false to save without validating.
      * @param fn optional callback
      */
-    save(fn?: (err: any, product: this, numAffected: number) => void): Promise<this>;
+    save(options?: SaveOptions, fn?: (err: any, product: this, numAffected: number) => void): Promise<this>;
```

Comment:
@fabiob I tried this out and it looks like this messes up the types in the callback fn:
```js
model.save(cb);    // Typescript thinks cb is a SaveOptions object instead of a function
```
Partly my fault. The tests don't do a good job of catching this.

Can you make it 2 lines like this?
```js
save(options?: SaveOptions, fn?: (err: any, product: this, numAffected: number) => void): Promise<this>;
save(fn?: (err: any, product: this, numAffected: number) => void): Promise<this>;
```